runtime/js-compute-runtime/builtins/cache-core.cpp

```diff
@@ -0,0 +1,1162 @@
+#include "cache-core.h"
+#include "body.h"
+#include "builtin.h"
+#include "builtins/native-stream-source.h"
+#include "core/encode.h"
+#include "host_interface/host_api.h"
+#include "js-compute-builtins.h"
+#include "js/Stream.h"
+#include <iostream>
+
+namespace builtins {
+
+namespace {
+
+// The JavaScript LookupOptions parameter we are parsing should have the below interface:
+// interface LookupOptions {
+//   headers?: HeadersInit;
+// }
+JS::Result<host_api::CacheLookupOptions> parseLookupOptions(JSContext *cx,
+                                                            JS::HandleValue options_val) {
+  host_api::CacheLookupOptions options;
+  if (!options_val.isUndefined()) {
+    if (!options_val.isObject()) {
+      JS_ReportErrorASCII(cx, "options argument must be an object");
+      return JS::Result<host_api::CacheLookupOptions>(JS::Error());
+    }
+    JS::RootedObject options_obj(cx, &options_val.toObject());
+    JS::RootedValue headers_val(cx);
+    if (!JS_GetProperty(cx, options_obj, "headers", &headers_val)) {
+      return JS::Result<host_api::CacheLookupOptions>(JS::Error());
+    }
+    // headers property is optional
+    if (!headers_val.isUndefined()) {
+      JS::RootedObject headersInstance(
+          cx,
+          JS_NewObjectWithGivenProto(cx, &builtins::Headers::class_, builtins::Headers::proto_obj));
+      if (!headersInstance) {
+        return JS::Result<host_api::CacheLookupOptions>(JS::Error());
+      }
+      auto headers = builtins::Headers::create(
+          cx, headersInstance, builtins::Headers::Mode::Standalone, nullptr, headers_val, true);
+      if (!headers) {
+        return JS::Result<host_api::CacheLookupOptions>(JS::Error());
+      }
+      JS::RootedValue headers_val(cx, JS::ObjectValue(*headers));
+      JS::RootedObject requestInstance(cx, Request::create_instance(cx));
+      if (!requestInstance) {
+        return JS::Result<host_api::CacheLookupOptions>(JS::Error());
+      }
+
+      JS::RootedValue input(cx, JS::StringValue(JS_NewStringCopyZ(cx, "http://example.com")));
+      JS::RootedObject request(cx,
+                               builtins::Request::create(cx, requestInstance, input, headers_val));
+      options.request_headers = host_api::HttpReq(builtins::Request::request_handle(request));
```

Comment:
Could you add a comment about why `http://example.com` is used here?

Reply:
Good point, I've now added one in [`38a8ce9` (#566)](https://github.com/fastly/js-compute-runtime/pull/566/commits/38a8ce9612e3af8081304979ee6983d5cfa679c8)